From f80e38380d34c843539f7ad4db1d65225e0d0c79 Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Fri, 3 Jun 2016 17:06:18 -0400
Subject: [PATCH 0183/1722] drm/amdgpu: work around lack of upstream ACPI
 support for D3cold

Until Dave's patch to support the new hybrid gfx ACPI method goes
upstream, we can fallback to the old ATPX method which seems to
still work.

Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
Signed-off-by: Kalyan Alle <kalyan.alle@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_atpx_handler.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_atpx_handler.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_atpx_handler.c
index 313169a..8988eb6 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_atpx_handler.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_atpx_handler.c
@@ -198,7 +198,17 @@ static int amdgpu_atpx_validate(struct amdgpu_atpx *atpx)
  
         if (valid_bits & ATPX_MS_HYBRID_GFX_SUPPORTED) {
                 printk("Hybrid Graphics, ATPX dGPU power cntl disabled\n");
+#if 1
+                /* This is a temporary hack until the D3 cold support
+                 * makes it upstream.  The ATPX power_control method seems
+                 * to still work on even if the system should be using
+                 * the new standardized hybrid D3 cold ACPI interface.
+                 */
+                atpx->functions.power_cntl = true;
+#else
                 atpx->functions.power_cntl = false;
+#endif
+		atpx->is_hybrid = true;
         }
 
 
-- 
2.7.4

